plugins/spi/SPIOutput.cpp

```diff
@@ -909,31 +909,39 @@ void SPIOutput::IndividualWS2812bControl(const DmxBuffer &buffer) {
     return;
   }
 
-  const unsigned int length = std::min(m_pixel_count * WS2812B_SLOTS_PER_PIXEL,
-                                       buffer.Size() - first_slot);
+  const unsigned int length = m_pixel_count;
```

Comment:
You need to keep the min to not over-read the buffer, otherwise if the DMX buffer is say 10 slots long in total, and you try to read data for four pixels, g and b of pixel 4 won't be in the buffer!

Reply:
When you say initialise, do you mean to send the o444444 out to all of them?